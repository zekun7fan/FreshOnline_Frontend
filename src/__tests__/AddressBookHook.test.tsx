import {renderHook, act, cleanup} from '@testing-library/react-hooks';

import AddressBookUI, {AddressBook} from "../components/AddressBook";


afterEach(() => {
    cleanup().then()
})

test('render address book', () => {
    const { result } = renderHook(() => AddressBook())
    const { setData, componentDidMount, ApplyChange,
        pushUpdate, EditLocation, deleteLocation, addLine, textChange, getElements } = result.current
    act(() => {setData([{"row":1, "alive":true, "editing":true, "text":"hello"}])})
    act(() => {componentDidMount()})
    setTimeout(() => {
        act(() => {ApplyChange(0)})
        act(() => {pushUpdate()})
        act(() => {EditLocation(0)})
        act(() => {deleteLocation(0)})
        act(() => {addLine()})
        act(() => {getElements()})
    }, 3000)
})

test('render address book UI', () => {
    const { result } = renderHook(() => AddressBookUI())
    expect(result.current).toMatchSnapshot()
})