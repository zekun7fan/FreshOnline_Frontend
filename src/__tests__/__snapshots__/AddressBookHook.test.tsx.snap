// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render address book UI 1`] = `
<div>
  <Button
    onClick={[Function]}
  >
    Add
  </Button>
</div>
`;
